test/integration/docker.el

```diff
@@ -0,0 +1,47 @@
+(require 'test/common)
+
+(defvar eldev--docker-emacs-version "27.2")
+
+(ert-deftest eldev-docker-emacs-1 ()
+  (skip-unless (and (eldev-docker-executable nil)
+                    (eldev--docker-on-supported-os)))
+  (let ((eldev--test-eldev-local (expand-file-name default-directory)))
+    (eldev--test-run "trivial-project"
+        ("--quiet"
+         "docker"
+         eldev--docker-emacs-version
+         "emacs"
+         "--batch"
+         "--eval"
+         `(prin1 (+ 1 2)))
+        (should (string-suffix-p "3" stdout))
```

Comment:
The stdout is exactly the same as the command w/o docker if the user has the image before hand, but on CI the "not found locally, pulling bla..." will be on the start of stdout.  If you don't like this I can see what can be done, maybe filtering it out in the forwarded output or something.